lib/rules/prefer-object-spread.js

```diff
@@ -196,6 +196,34 @@ function autofixObjectLiteral(node, sourceCode) {
     };
 }
 
+/**
+ * Check if the node has modified a given variable
+ * @param {ASTNode|null} node - The node that the rule warns on, i.e. the Object.assign call
+ * @returns {boolean} - true if node is an assignment, variable declaration, or import statement
+ */
+function isModifier(node) {
+    if (!node.type) {
+        return false;
+    }
+
+    return node.type === "AssignmentExpression" ||
+        node.type === "VariableDeclarator" ||
+        node.type === "ImportDeclaration";
+}
+
+/**
+ * Check if the node has modified a given variable
+ * @param {array} references - list of reference nodes
+ * @returns {boolean} - true if node is has been overwritten by an assignment or import
+ */
+function modifyingObjectReference(references) {
+    return references.some(ref => (
+        ref.identifier &&
+        ref.identifier.parent &&
```

Comment:
Ack, I missed this one. Same indent issue here, would you mind indenting this line and the next?

Reply:
I'm a little unclear on what the correct indentation is for this, how is this one supposed to be written? the function code is:
```js
function modifyingObjectReference(references) {
    return references.some(ref => (
        ref.identifier &&
        ref.identifier.parent &&
        isModifier(ref.identifier.parent)
    ));
}
```

Reply:
Hmm. What I had in mind was:

```js
function modifyingObjectReference(references) {
    return references.some(ref => (
        ref.identifier &&
            ref.identifier.parent &&
            isModifier(ref.identifier.parent)
    ));
}
```

Looking back, I don't mind this one as much since the lines are already indented from statement position, so nobody will confuse these as potentially multiple statements. If you think this is a reasonable change, please make it; but if not, I can wait for others to review and go with consensus.